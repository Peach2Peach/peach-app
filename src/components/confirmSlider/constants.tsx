import tw from '../../styles/tailwind'

export const padding = tw`w-1`.width as number
export const trackWidth = 260
export const knobWidth = tw`w-18`.width as number
