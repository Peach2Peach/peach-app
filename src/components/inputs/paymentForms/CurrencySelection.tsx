import { View } from "react-native";
import tw from "../../../styles/tailwind";
import i18n from "../../../utils/i18n";
import { getPaymentMethodInfo } from "../../../utils/paymentMethod/getPaymentMethodInfo";
import { PeachText } from "../../text/PeachText";
import { CurrencyItem } from "../CurrencyItem";

type Props = {
  paymentMethod: PaymentMethod;
  selectedCurrencies: Currency[];
  onToggle: (currencies: Currency) => void;
};

export const CurrencySelection = ({
  paymentMethod,
  selectedCurrencies,
  onToggle,
}: Props) => (
  <View>
    <View style={tw`flex-row items-center`}>
      <PeachText style={tw`input-label`}>
        {i18n("form.additionalCurrencies")}
      </PeachText>
    </View>
    <View style={tw`flex-row flex-wrap gap-2 mt-1`}>
      {getPaymentMethodInfo(paymentMethod)?.currencies.map((currency) => (
        <CurrencyItem
          key={currency}
          label={currency}
          isSelected={selectedCurrencies.includes(currency)}
          onPress={() =>
            !selectedCurrencies.includes(currency) ||
            selectedCurrencies.length > 1
              ? onToggle(currency)
              : null
          }
        />
      ))}
    </View>
  </View>
);
