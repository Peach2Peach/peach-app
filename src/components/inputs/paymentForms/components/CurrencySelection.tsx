import { View } from "react-native";
import tw from "../../../../styles/tailwind";
import { getPaymentMethodInfo } from "../../../../utils/paymentMethod/getPaymentMethodInfo";
import { PeachText } from "../../../text/PeachText";
import { CurrencyItem } from "../../CurrencyItem";
import { useTranslate } from "@tolgee/react";

type Props = ComponentProps & {
  paymentMethod: PaymentMethod;
  selectedCurrencies: Currency[];
  onToggle: (currencies: Currency) => void;
};

export const CurrencySelection = ({
  paymentMethod,
  selectedCurrencies,
  onToggle,
  style,
}: Props) => {
  const { t } = useTranslate("form");

  return (
    <View style={style}>
      <View style={tw`flex-row items-center`}>
        <PeachText style={tw`input-label`}>
          {t("form.additionalCurrencies")}
        </PeachText>
      </View>
      <View style={tw`flex-row flex-wrap gap-2 mt-1`}>
        {getPaymentMethodInfo(paymentMethod)?.currencies.map((currency) => (
          <CurrencyItem
            key={currency}
            label={currency}
            isSelected={selectedCurrencies.includes(currency)}
            onPress={() =>
              !selectedCurrencies.includes(currency) ||
              selectedCurrencies.length > 1
                ? onToggle(currency)
                : null
            }
          />
        ))}
      </View>
    </View>
  );
};
