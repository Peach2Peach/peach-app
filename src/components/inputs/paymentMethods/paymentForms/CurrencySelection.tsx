import React, { ReactElement, useContext, useState } from 'react'
import { Pressable, View } from 'react-native'
import { OverlayContext } from '../../../../contexts/overlay'
import Currency from '../../../../overlays/info/Currency'
import tw from '../../../../styles/tailwind'
import i18n from '../../../../utils/i18n'
import { getPaymentMethodInfo } from '../../../../utils/paymentMethod'
import Icon from '../../../Icon'
import { Text } from '../../../text'
import { Item } from '../../Item'

export const toggleCurrency = (currency: Currency) => (currencies: Currency[]) => {
  if (currencies.indexOf(currency) === -1) {
    currencies.push(currency)
  } else {
    currencies = currencies.filter(c => c !== currency)
  }
  return [...currencies]
}

type CurrencySelectionProps = ComponentProps & {
  paymentMethod: PaymentMethod
  selectedCurrencies: Currency[]
  onToggle: (currencies: Currency) => void
}

export const CurrencySelection = ({
  paymentMethod,
  selectedCurrencies,
  onToggle,
  style,
}: CurrencySelectionProps): ReactElement => {
  const [, updateOverlay] = useContext(OverlayContext)
  const [paymentMethodInfo] = useState(getPaymentMethodInfo(paymentMethod))

  const openCurrencyHelp = () =>
    updateOverlay({
      content: <Currency />,
      help: true,
      showCloseIcon: true,
    })

  return (
    <View style={style}>
      <View style={tw`items-center flex-row`}>
        <Text style={tw`font-baloo text-lg`}>
          {i18n('form.additionalCurrencies')} ({i18n('form.optional')})
        </Text>
        <Pressable style={tw`p-3`} onPress={openCurrencyHelp}>
          <Icon id="help" style={tw`w-5 h-5`} color={tw`text-blue-1`.color as string} />
        </Pressable>
      </View>
      <View style={tw`flex-row mt-1`}>
        {paymentMethodInfo.currencies.map((currency, i) => (
          <Item
            key={currency}
            style={[tw`h-6 px-2`, i > 0 ? tw`ml-2` : {}]}
            label={currency}
            isSelected={selectedCurrencies.indexOf(currency) !== -1}
            onPress={() =>
              selectedCurrencies.indexOf(currency) === -1 || selectedCurrencies.length > 1 ? onToggle(currency) : null
            }
          />
        ))}
      </View>
    </View>
  )
}
