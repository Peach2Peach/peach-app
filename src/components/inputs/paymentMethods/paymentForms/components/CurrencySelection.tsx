import { useState } from 'react'
import { View } from 'react-native'
import tw from '../../../../../styles/tailwind'
import i18n from '../../../../../utils/i18n'
import { getPaymentMethodInfo } from '../../../../../utils/paymentMethod'
import { Text } from '../../../../text'
import { CurrencyItem } from '../../../CurrencyItem'

type Props = ComponentProps & {
  paymentMethod: PaymentMethod
  selectedCurrencies: Currency[]
  onToggle: (currencies: Currency) => void
}

export const CurrencySelection = ({ paymentMethod, selectedCurrencies, onToggle, style }: Props) => {
  const [paymentMethodInfo] = useState(getPaymentMethodInfo(paymentMethod))

  return (
    <View style={style}>
      <View style={tw`flex-row items-center`}>
        <Text style={tw`input-label`}>{i18n('form.additionalCurrencies')}</Text>
      </View>
      <View style={tw`flex-row flex-wrap gap-2 mt-1`}>
        {paymentMethodInfo.currencies.map((currency) => (
          <CurrencyItem
            key={currency}
            label={currency}
            isSelected={selectedCurrencies.includes(currency)}
            onPress={() =>
              !selectedCurrencies.includes(currency) || selectedCurrencies.length > 1 ? onToggle(currency) : null
            }
          />
        ))}
      </View>
    </View>
  )
}
