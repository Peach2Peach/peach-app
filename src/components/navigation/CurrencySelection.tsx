import { Pressable, View } from 'react-native'
import { Text } from '..'
import tw from '../../styles/tailwind'

type CurrencySelectionItemProps = ComponentProps & {
  currency: Currency
  isSelected: boolean
  onPress: (currency: Currency) => void
}
const CurrencySelectionItem = ({ currency, isSelected, onPress, style }: CurrencySelectionItemProps) => (
  <Pressable style={style} onPress={() => onPress(currency)}>
    <Text numberOfLines={1} style={[tw`text-center button-large text-black-2`, isSelected && tw`text-black-1`]}>
      {currency}
    </Text>
    {isSelected && <View style={[tw`w-full h-0.5 -mt-0.5 bg-black-1 rounded-1px`]} />}
  </Pressable>
)

const ItemSeparator = ({ style }: ComponentProps) => <View style={[tw`w-px h-4 bg-black-6 rounded-1px`, style]} />

type Props = ComponentProps & {
  currencies: Currency[]
  selected: Currency
  select: (currency: Currency) => void
}

export const CurrencySelection = ({ currencies, selected, select, style }: Props) => (
  <View style={[tw`flex-row flex-wrap justify-center`, style]}>
    {currencies.map((currency, index) => (
      <View style={tw`flex-row items-center flex-grow min-w-1/8 max-w-1/4`} key={`currency-selection-${currency}`}>
        <CurrencySelectionItem
          currency={currency}
          isSelected={currency === selected}
          onPress={select}
          style={tw`flex-grow`}
        />
        {index < currencies.length - 1 && (
          <ItemSeparator key={`currency-selection-separator${currency}`} style={tw`mx-1`} />
        )}
      </View>
    ))}
  </View>
)
