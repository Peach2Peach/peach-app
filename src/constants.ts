import { getBuildNumber, getUniqueId, getVersion, isEmulatorSync } from 'react-native-device-info'
import { unique } from './utils/array'
import { sha256 } from './utils/crypto/sha256'

export const SATSINBTC = 100000000
export const MSINADAY = 86400000

export let PEACHPGPPUBLICKEY = ''
export const setPeachPGPPublicKey = (pgpPublicKey: string) => (PEACHPGPPUBLICKEY = pgpPublicKey)

export let PEACHFEE = 0.015
export const setPeachFee = (fee: number) => (PEACHFEE = fee)

export const MAXMININGFEE = 20000

// time go automatically restart app when calling app from background after this time has passed
export const TIMETORESTART = 1000 * 60 * 5

export const APPVERSION = getVersion()
export const BUILDNUMBER = getBuildNumber()

export let CLIENTSERVERTIMEDIFFERENCE = 0
export const setClientServerTimeDifference = (diff: number) => (CLIENTSERVERTIMEDIFFERENCE = diff)

export const ISEMULATOR = isEmulatorSync()

export const UNIQUEID = sha256(getUniqueId())

export let MINAPPVERSION = APPVERSION
export const setMinAppVersion = (ver: string) => (MINAPPVERSION = ver)

export let LATESTAPPVERSION = APPVERSION
export const setLatestAppVersion = (ver: string) => (LATESTAPPVERSION = ver)

export let CURRENCIES: Currency[] = ['EUR', 'CHF', 'GBP', 'SEK']

export let COUNTRIES: Country[] = ['DE', 'FR', 'IT', 'ES', 'NL', 'UK', 'SE']

export let PAYMENTMETHODS: PaymentMethod[] = ['sepa']
export let PAYMENTMETHODINFOS: PaymentMethodInfo[] = [
  {
    id: 'sepa',
    currencies: ['EUR'],
    annonymous: false,
  },
]

export const PAYMENTCATEGORIES: PaymentCategories = {
  bankTransfer: ['sepa'],
  onlineWallet: ['paypal', 'revolut', 'wise', 'twint', 'swish'],
  giftCard: ['giftCard.amazon'].concat(COUNTRIES.map((c) => `giftCard.amazon.${c}`)) as PaymentMethod[],
  localOption: ['mbWay', 'bizum', 'satispay'],
  cash: ['cash'],
  cryptoCurrency: [],
}

export const LOCALPAYMENTMETHODS: LocalPaymentMethods = {
  EUR: {
    IT: ['satispay'],
    PT: ['mbWay'],
    ES: ['bizum'],
  },
}

export const APPLINKS: Partial<Record<PaymentMethod, { appLink?: string; url: string; userLink?: string }>> = {
  paypal: {
    url: 'https://paypal.com/open_web',
    userLink: 'https://paypal.com/paypalme/',
  },
  revolut: {
    url: 'https://revolut.com/app',
    userLink: 'https://revolut.me/',
  },
  satispay: { url: 'https://satispay.com/app' },
  wise: { url: 'https://wise.com/user/account' },
  'giftCard.amazon.DE': { url: 'https://www.amazon.de/dp/B0B2Q4ZRDW' },
  'giftCard.amazon.FR': { url: 'https://www.amazon.fr/dp/B004MYH1YI' },
  'giftCard.amazon.IT': { url: 'https://www.amazon.it/dp/B07DWRCL44' },
  'giftCard.amazon.ES': { url: 'https://www.amazon.es/dp/B07H8STZ9N' },
  'giftCard.amazon.NL': { url: 'https://www.amazon.nl/dp/B07W6D728D' },
  'giftCard.amazon.UK': { url: 'https://www.amazon.co.uk/dp/B07S6C1DZ6' },
  'giftCard.amazon.SE': { url: 'https://www.amazon.se/dp/B089VNKFM7' },
}

export const setPaymentMethods = (paymentMethodInfos: PaymentMethodInfo[]) => {
  PAYMENTMETHODINFOS = paymentMethodInfos
  CURRENCIES = paymentMethodInfos.reduce((arr, info) => arr.concat(info.currencies), [] as Currency[]).filter(unique())
  COUNTRIES = paymentMethodInfos
    .reduce((arr, info) => arr.concat(info.countries || []), [] as Country[])
    .filter(unique())
  PAYMENTMETHODS = paymentMethodInfos.map((method) => method.id)
}

export let MINTRADINGAMOUNT = 200000
export let MAXTRADINGAMOUNT = 5000000
export const setMinTradingAmount = (amount: number) => (MINTRADINGAMOUNT = amount)
export const setMaxTradingAmount = (amount: number) => (MAXTRADINGAMOUNT = amount)

type Timers = {
  [key in ContractAction]: number
}

// Time in ms
export const TIMERS: Timers = {
  none: 0,
  kycResponse: 1000 * 60 * 60 * 12,
  sendPayment: 1000 * 60 * 60 * 12,
  confirmPayment: 1000 * 60 * 60 * 12,
}

// Reputation
export const GOLDMEDAL = 0.9
export const SILVERMEDAL = 0.7

export const MEDALS = ['fastTrader', 'superTrader', 'ambassador']
