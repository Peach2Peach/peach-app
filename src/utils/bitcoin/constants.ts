import { Transaction } from 'bitcoinjs-lib'

export const SIGHASH = {
  ALL: Transaction.SIGHASH_ALL,
  SINGLE_ANYONECANPAY: Transaction.SIGHASH_SINGLE + Transaction.SIGHASH_ANYONECANPAY,
}
