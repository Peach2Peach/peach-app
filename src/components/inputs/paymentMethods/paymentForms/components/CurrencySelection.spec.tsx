import { CurrencySelection } from './CurrencySelection'
import { createRenderer } from 'react-test-renderer/shallow'
import { setPaymentMethods } from '../../../../../constants'
import { fireEvent, render } from '@testing-library/react-native'

describe('CurrencySelection', () => {
  const renderer = createRenderer()
  const onToggle = jest.fn()

  setPaymentMethods([
    {
      id: 'revolut',
      currencies: ['EUR', 'CHF', 'GBP', 'BGN', 'CZK', 'DKK', 'HUF', 'ISK', 'NOK', 'PLN', 'RON', 'SEK'],
      anonymous: false,
    },
  ])
  it('should render correctly', () => {
    renderer.render(
      <CurrencySelection paymentMethod="revolut" selectedCurrencies={['EUR', 'CHF']} onToggle={onToggle} />,
    )
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
  it('should toggle a selection correctly', () => {
    const { getByText } = render(
      <CurrencySelection paymentMethod="revolut" selectedCurrencies={['EUR', 'CHF']} onToggle={onToggle} />,
    )
    fireEvent(getByText('GBP'), 'onPress')
    expect(onToggle).toHaveBeenCalledWith('GBP')
  })
})
