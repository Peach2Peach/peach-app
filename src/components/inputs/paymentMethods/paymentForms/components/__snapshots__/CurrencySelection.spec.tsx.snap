// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencySelection renders correctly 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <PeachText
      style={
        {
          "fontFamily": "Baloo2-Bold",
          "fontSize": 16,
          "letterSpacing": 0.16,
          "lineHeight": 20,
        }
      }
    >
      Which currencies do you want toÂ accept?
    </PeachText>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "gap": 8,
        "marginTop": 4,
      }
    }
  >
    <CurrencyItem
      isSelected={true}
      label="EUR"
      onPress={[Function]}
    />
    <CurrencyItem
      isSelected={true}
      label="CHF"
      onPress={[Function]}
    />
    <CurrencyItem
      isSelected={false}
      label="GBP"
      onPress={[Function]}
    />
    <CurrencyItem
      isSelected={false}
      label="BGN"
      onPress={[Function]}
    />
    <CurrencyItem
      isSelected={false}
      label="CZK"
      onPress={[Function]}
    />
    <CurrencyItem
      isSelected={false}
      label="DKK"
      onPress={[Function]}
    />
    <CurrencyItem
      isSelected={false}
      label="HUF"
      onPress={[Function]}
    />
    <CurrencyItem
      isSelected={false}
      label="ISK"
      onPress={[Function]}
    />
    <CurrencyItem
      isSelected={false}
      label="NOK"
      onPress={[Function]}
    />
    <CurrencyItem
      isSelected={false}
      label="PLN"
      onPress={[Function]}
    />
    <CurrencyItem
      isSelected={false}
      label="RON"
      onPress={[Function]}
    />
    <CurrencyItem
      isSelected={false}
      label="SEK"
      onPress={[Function]}
    />
  </View>
</View>
`;
